### PR TITLE
feat(auth): show PasskeyPromptDialog for passkey requests

### PR DESCRIPTION
**Pull request by ice-tychon**
## Description
This PR shows `PasskeyPromptDialog` when the app asks a user for a Passkey confirmation. Currently it's implemented only for login, register, and a few other pages.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactoring
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

File: lib/app/components/auth/passkey_prompt_dialog.dart [hunk 1]
```diff
@@ -7,13 +7,31 @@ import 'package:ion/app/components/progress_bar/ion_loading_indicator.dart';
...
 import 'package:ion/app/features/core/providers/theme_mode_provider.dart';
+import 'package:ion/app/hooks/use_on_init.dart';
 import 'package:ion/generated/assets.gen.dart';
 
-class PasskeysPromptDialog extends ConsumerWidget {
-  const PasskeysPromptDialog({super.key});
+class PasskeyPromptDialog extends HookConsumerWidget {
+  const PasskeyPromptDialog({
+    required this.onInit,
+    required this.provider,
```
Comment (ice-alcides) [inline lib/app/components/auth/passkey_prompt_dialog.dart:16]: don't pass a provider as a widget parameter (https://riverpod.dev/docs/essentials/do_dont). Use `Ref` or pass a callback, or use a riverpod-provider directly in `ref.listen`

Comment (ice-alcides) [reply lib/app/components/auth/passkey_prompt_dialog.dart]: discussed

File: lib/app/components/auth/passkey_prompt_dialog.dart [hunk 2]
```diff
@@ -9,8 +9,10 @@ import 'package:ion/app/extensions/extensions.dart';
 import 'package:ion/app/features/core/providers/theme_mode_provider.dart';
 import 'package:ion/generated/assets.gen.dart';
 
-class PasskeysPromptDialog extends ConsumerWidget {
-  const PasskeysPromptDialog({super.key});
+class PasskeyPromptDialog extends HookConsumerWidget {
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply lib/app/components/auth/passkey_prompt_dialog.dart]: Fixed

File: lib/app/components/auth/passkey_prompt_dialog_helper.dart
[the remark this answers is not in the document]
Comment (ice-tychon) [reply lib/app/components/auth/passkey_prompt_dialog_helper.dart:1]: Fixed

**Pull request merged**